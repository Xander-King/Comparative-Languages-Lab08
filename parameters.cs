
namespace Params {
    class Program {
        // TODO: Write a static 'swap' method
        //       The method takes two parameters and excahnges their values. 


        // TODO: Write a static bool TryParseInt  function that 
        //       returns true of it successfully reads an integer from
        //       the Console and fals otherwise.  The integer should be
        //       returned via a parameter to the function. 
        //       
        //       Use Console.ReadLine to read a string
        //       Use Int32.Parse to convert the strin to an integer
        //       Use a Try/Catch block to handle one of the documented exceptions
        //       Please refer to the docs for Int32.Parse to see the list of exceptions to handle. 


        public static void Main() {
            int a, b;
            
            Console.WriteLine("Please enter a number for a: ")
            while (not TryParseInt(a)) { //<-- You will need to change something here
                Console.WriteLine("ERROR - Invalid number.")
                Console.Write("Please enter a number for a:")
            }


            Console.WriteLine("Please enter number for b : ")
            while (not TryParseInt(out b)) { //<-- You will need to change somehing here
                Console.WriteLine("ERROR - Invalid number.")
                Console.Write("Please enter a number for b:")
            }

            Console.WriteLine($"Before: a={a, -6} and b={b,-6}");
            
            swap(a, b); //<-- You will need to make a change to this line

            Console.WriteLine($"After : a={a, -6} and b={b,-6}");
        }
    }
}