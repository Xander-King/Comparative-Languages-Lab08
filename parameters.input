a
b
notanumber
10
z
x 
stillnotanumber
15