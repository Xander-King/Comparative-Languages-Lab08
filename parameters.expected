
Please enter a number for a: 
ERROR - Invalid number.
Please enter a number for a:
ERROR - Invalid number.
Please enter a number for a:
ERROR - Invalid number.
Please enter a number for a:
a has been set to 10

Please enter number for b : 
ERROR - Invalid number.
Please enter a number for b:
ERROR - Invalid number.
Please enter a number for b:
ERROR - Invalid number.
Please enter a number for b:
b has been set to 15

Before: a=10     and b=15    
After : a=15     and b=10    
